src/caffe/solver.cpp

```diff
@@ -775,9 +776,112 @@ void AdaGradSolver<Dtype>::ComputeUpdateValue(int param_id, Dtype rate) {
   }
 }
 
+
+template <typename Dtype>
+void AdamSolver<Dtype>::ComputeUpdateValue(int param_id, Dtype rate) {
+  const vector<shared_ptr<Blob<Dtype> > >& net_params = this->net_->params();
+  const vector<float>& net_params_lr = this->net_->params_lr();
+  Dtype local_rate = rate * net_params_lr[param_id];
+  const Dtype beta1 = this->param_.momentum();
+  const Dtype beta2 = this->param_.momentum2();
+
+  const int t = this->iter_ / this->param_.stepsize() +1;
```

Comment:
param_.stepsize() should not be used for this solver at all, it's usage is for the step lr_policy, using it here only makes it unclear what the alpha/stepsize from the paper is.

It does need the +1 to do the (1-beta1^t) correctly, so it should be:
const int t = this->iter_ + 1;